Security/Principal/WindowsLoggedInIdentity.cs

```diff
@@ -79,9 +93,13 @@ public WindowsLoggedInIdentity(WindowsIdentity identityToImpersonate)
 		/// <summary>Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.</summary>
 		public virtual void Dispose()
 		{
-			if (ownId) AuthenticatedIdentity?.Dispose();
+			if (ownId)
```

Comment:
This is actually the spot that need fixing. I think `AuthenticatedIdentity` should only be nulled when `ownId` = `true`.

Reply:
`AuthenticatedIdentity` needs to be set to `null` for both constructors so it can be garbage collected.

Reply:
Yes, but in the original code, `AuthenticatedIdentity` always gets nulled, even if we don't own it.

Reply:
`AuthenticatedIdentity` is a property value that holds the instance either created by the constructor or passed into the constructor. All I'm doing is effectively releasing the reference.